src/itkOMEZarrNGFFImageIO.cxx

```diff
@@ -281,8 +281,16 @@ OMEZarrNGFFImageIO::ReadArrayMetadata(std::string path)
   this->SetComponentType(tensorstoreToITKComponentType(dtype));
 
   std::vector<int64_t> dims(shape_span.rbegin(), shape_span.rend()); // convert KJI into IJK
-  assert(this->GetNumberOfDimensions() == dims.size());
-  // this->SetNumberOfDimensions(dims.size());
+
+  if (this->GetNumberOfDimensions() == 0) // reading version 0.2 or 0.1
+  {
+    this->InitializeIdentityMetadata(dims.size());
+  }
+  else
+  {
+    assert(this->GetNumberOfDimensions() == dims.size());
```

Comment:
Not sure if it matters, but could use `itkAssertInDebugOrThrowInReleaseMacro` here instead of `assert`.

Reply:
That has a different syntax:
```cpp
if (condition)
{
  itkAssertInDebugOrThrowInReleaseMacro(<<"Message");
}
```
which is a bit cumbersome for these sanity checks.

Reply:
Maybe you meant `itkAssertInDebugAndIgnoreInReleaseMacro`? That is a simple wrapper around `assert`.

Reply:
Makes sense, good to leave as is.